sui_programmability/adapter/src/adapter.rs

```diff
@@ -661,6 +663,7 @@ pub fn resolve_and_type_check(
         });
     }
 
+    // TODO: not true anymore
     // total number of args is (|objects| + |pure_args|) + 1 for the the `TxContext` object
```

Comment:
While we have a fixed # of args, we can have any number of objects with vectors, so we need to rework some of the logic here